pkg/controller/tas/resource_flavor.go

```diff
@@ -43,6 +47,34 @@ import (
 	"sigs.k8s.io/kueue/pkg/queue"
 )
 
+var nodeSemantic = conversion.EqualitiesOrDie(
+	nodeConditionEqual,
+	// Handle metav1.Time comparison to avoid panic on unexported fields
+	func(a, b metav1.Time) bool {
+		return a.Time.Equal(b.Time)
+	},
+	// Handle *metav1.Time pointer comparison (e.g., Taint.TimeAdded)
+	func(a, b *metav1.Time) bool {
+		if a == nil && b == nil {
+			return true
+		}
+		if a == nil || b == nil {
+			return false
+		}
+		return a.Time.Equal(b.Time)
+	},
+	// Handle resource.Quantity comparison to avoid panic on unexported fields
+	func(a, b resource.Quantity) bool {
+		return a.Cmp(b) == 0
```

Comment:
```suggestion
		return a.Equal(b)
```

We have `Equal` function.